### ISSUE TITLE
Questions about data alignment in the GECKO exercise

### ISSUE THREAD

**Issue by Weiming-Hu** [NCAR/ai4ess-hackathon-2020#27]
I have encountered some confusion when trying to understand [this line](https://github.com/NCAR/ai4ess-hackathon-2020/blame/master/notebooks/gecko.ipynb#L942), in particular **why we need `iloc[1:,:]` to remove the first element in the predictions**. Similarly, on the next line, why do we need to remove the last element from the true values?

My understanding is that the NN is trained with already aligned datasets. The datasets created from the function `prepare_data` already have an offset by 1 to compensate for the purpose of "predicting the next timestamp". When using the model on validation set, why do we still need to remove any elements?

I have tried to remove the two `iloc` places on [line 942](https://github.com/NCAR/ai4ess-hackathon-2020/blame/master/notebooks/gecko.ipynb#L942) and [line 943](https://github.com/NCAR/ai4ess-hackathon-2020/blame/master/notebooks/gecko.ipynb#L943). But I found my errors went up so I suppose the original code would be correct. But why?

I hope you can help me understand the code a bit better.

Thank you

**Comment by yang0920colostate** [comment]
I can try to answer this question.
It has to do with the NN coupling with the Box Emulator Model.
From the description in the code: _This model (box_emulator()), takes one timestep of initial conditions to feed the neural network we built. It then predicts one timestep, and uses the prediction as the input for another prediction (along with the static environmental variables), and so forth for n timesteps._
As a result, from the Box Emulator point of view, the first "prediction" comes after you use the prediction from inputting row = 0 into the NN. That's why we remove the first element because it is the initial state (not a prediction for the Box Emulator Model).

**Comment by Weiming-Hu** [comment]
Thank you @yang0920colostate . I'm cloing this ticket.
